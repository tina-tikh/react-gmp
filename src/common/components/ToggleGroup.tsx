import * as React from 'react';
import {ReactNode} from 'react';

import styled from '../styled';
import Button from './Button';

type ToggleGroupProps = {
    label: string,
    opts: string[],
    value: string,
    onChange: (i: string) => void
}

type ToggleGroupState = {
    value: string
}

const Container = styled.div`
    display: inline-block;
    font-size: 1.6rem;
    text-transform: uppercase;
`;
const Label = styled.label`
    margin-right: 2rem;
`;
const Toggle = styled(Button)`
    background-color: rgba(255, 255, 255, 0.2);
    
    & + button {
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
    }
    
    &:first-of-type {
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
    }
    
    &:focus {
        outline: none;
    }

    &.active {
        background-color: ${props => props.theme.colors.primary};
    }
`;

class ToggleGroup extends React.PureComponent<ToggleGroupProps, ToggleGroupState> {
    constructor(props: ToggleGroupProps) {
        super(props);
        this.state = {value: this.props.value};
    }

    render(): ReactNode {
        return <Container>
            <Label>{this.props.label}</Label>
            {
                this.props.opts.map((item, i) => <Toggle
                    key={item}
                    className={`${this.getClassName(item)}`}
                    onClick={() => this.handleOnClick(item)}
                >{item}</Toggle>)
            }
        </Container>;
    }

    getClassName(item: string): string {
        return item == this.state.value ? "active" : "";
    }

    handleOnClick(item: string): void {
        this.setState({
            value: item
        });
        this.props.onChange(item);
    }
}

export default ToggleGroup;
