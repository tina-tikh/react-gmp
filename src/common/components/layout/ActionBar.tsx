import { styled } from '../../theme';

const ActionBar = styled.section`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2rem 6rem;
  background-color: ${(props) => props.theme.colors.dark};
`;

export default ActionBar;
