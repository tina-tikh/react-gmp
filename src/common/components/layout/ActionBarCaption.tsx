import { styled } from '../../theme';

const ActionBarCaption = styled.span`
  font-size: 1.4rem;
  font-weight: bold;
`;

export default ActionBarCaption;
