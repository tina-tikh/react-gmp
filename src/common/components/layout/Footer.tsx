import * as React from 'react';

import { styled } from '../../theme';
import Logo from './Logo';

const FooterBlock = styled.footer`
  padding: 2.5rem;
  text-align: center;
  background-color: ${(props) => props.theme.colors.dark};
`;

class Footer extends React.PureComponent {
  render() {
    return (
      <FooterBlock>
        <Logo></Logo>
      </FooterBlock>
    );
  }
}

export default Footer;
