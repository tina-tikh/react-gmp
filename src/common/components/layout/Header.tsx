import * as React from 'react';

import { styled } from '../../theme';
import Logo from './Logo';

const HeaderBlock = styled.header`
  padding: 3rem 6rem;
  color: ${(props) => props.theme.colors.light};
  background-color: ${(props) => props.theme.colors.darkest};
`;

class Header extends React.PureComponent {
  render() {
    return (
      <HeaderBlock>
        <Logo></Logo>
        {this.props?.children}
      </HeaderBlock>
    );
  }
}

export default Header;
