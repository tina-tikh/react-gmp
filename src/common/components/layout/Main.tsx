import * as React from 'react';

import { styled } from '../../theme';

const Main = styled.main`
  flex: 1;
  color: ${(props) => props.theme.colors.light};
  background-color: ${(props) => props.theme.colors.darker};
`;

export default Main;
