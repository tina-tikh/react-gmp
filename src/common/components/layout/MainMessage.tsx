import { styled } from '../../theme';

const MainMessage = styled.h1`
  margin: 5rem;
  text-align: center;
  font-size: 4rem;
`;

export default MainMessage;
