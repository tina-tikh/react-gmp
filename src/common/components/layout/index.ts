import ActionBar from './ActionBar';
import ActionBarCaption from './ActionBarCaption';
import Footer from './Footer';
import Header from './Header';
import Logo from './Logo';
import Main from './Main';
import MainMessage from './MainMessage';
import Page from './Page';

export {
  ActionBar,
  ActionBarCaption,
  Footer,
  Header,
  Logo,
  Main,
  MainMessage,
  Page,
};
