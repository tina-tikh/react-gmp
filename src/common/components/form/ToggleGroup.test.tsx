import * as React from 'react';
import { mount, shallow } from 'enzyme';

import { Page } from '../layout';
import ToggleGroup from './ToggleGroup';

const opts = ['1', '2', '3'];
const value = '1';

describe('<ToggleGroup />', () => {
  it('should be rendered', () => {
    const component = shallow(
      <ToggleGroup label="Sort" opts={opts} value={value} />
    );
    expect(component).toMatchSnapshot();
  });

  it('should call onChange', () => {
    const onChange = jest.fn();
    const component = mount(
      <Page>
        <ToggleGroup
          label="Sort"
          opts={opts}
          value={value}
          onChange={onChange}
        />
      </Page>
    );
    component.find('button').at(0).simulate('click');
    expect(onChange).toHaveBeenCalled();
  });
});
