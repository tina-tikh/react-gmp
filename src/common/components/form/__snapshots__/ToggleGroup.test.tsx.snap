// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleGroup /> should be rendered 1`] = `
<Styled(div)>
  <Styled(label)>
    Sort
  </Styled(label)>
  <Styled(button)
    className="active"
    key="1"
    onClick={[Function]}
  >
    1
  </Styled(button)>
  <Styled(button)
    className=""
    key="2"
    onClick={[Function]}
  >
    2
  </Styled(button)>
  <Styled(button)
    className=""
    key="3"
    onClick={[Function]}
  >
    3
  </Styled(button)>
</Styled(div)>
`;
